airbyte-webapp/src/components/StateBlock/StateBlock.tsx

```diff
@@ -0,0 +1,169 @@
+import { useConfirmationModalService } from "hooks/services/ConfirmationModal";
+import { useGetConnectionState, useUpdateConnectionState } from "hooks/services/useConnectionHook";
+
+type IProps = {
+  connectionId: string;
+};
+
+const StateBlockComponent = styled(ContentCard)`
+  margin-top: 12px;
+  padding: 19px 20px 20px;
+  // display: flex;
+  align-items: top;
+  justify-content: space-between;
+`;
+
+const Text = styled.div`
+  margin-left: 20px;
+  font-size: 11px;
+  line-height: 13px;
+  color: ${({ theme }) => theme.greyColor40};
+  white-space: pre-line;
+`;
+
+const ErrorText = styled.div`
+  margin-left: 20px;
+  font-size: 11px;
+  line-height: 13px;
+  color: ${({ theme }) => theme.redColor};
+  white-space: pre-line;
+`;
+
+const codeStyle = {
+  fontFamily: '"Fira code", "Fira Mono", monospace',
+  fontSize: 12,
+  marginTop: "5px",
+  marginLeft: "10px",
+  marginBottom: "5px",
+};
+
+type Validation = {
+  valid: boolean;
+  message?: string;
+};
+
+const StateBlock: React.FC<IProps> = ({ connectionId }) => {
+  const [stateString, setStateString] = useState<string>(`// ...`);
+  const [loading, setLoading] = useState(false);
+  const [validation, setValidation] = useState<Validation>({ valid: true });
+  const { openConfirmationModal, closeConfirmationModal } = useConfirmationModalService();
+  const { mutateAsync: getState } = useGetConnectionState();
+  const { mutateAsync: updateState } = useUpdateConnectionState();
+
+  const loadState = async () => {
+    setLoading(true);
+    const state = await getState(connectionId);
+    if (state?.state) {
+      setStateString(formatState(state.state));
+      setLoading(false);
+    }
+  };
+
+  const saveState = async () => {
+    setLoading(true);
+    const stateObject = JSON.parse(stateString) as ConnectionState;
+    const newState = await updateState({ connectionId, state: stateObject });
+    if (newState?.state) {
+      setStateString(formatState(newState.state));
+      setLoading(false);
+    }
+  };
+
+  const onSaveButtonClick = useCallback(() => {
+    const validationResponse = validateState(stateString);
+    setValidation(validationResponse);
+    if (!validationResponse.valid) return;
+
+    openConfirmationModal({
+      text: `tables.State.confirmModalText`,
+      title: `tables.State.confirmModalTitle`,
+      submitButtonText: "form.save",
+      onSubmit: async () => {
+        await saveState();
+        closeConfirmationModal();
+      },
+      submitButtonDataId: "state",
+    });
+    // eslint-disable-next-line
+  }, [closeConfirmationModal, saveState, openConfirmationModal]);
+
+  const loadStateMemoized = useCallback(() => {
+    loadState();
+    // eslint-disable-next-line
+  }, []);
+
+  useEffect(() => {
+    loadStateMemoized();
+  }, [loadStateMemoized]);
+
+  return (
+    <StateBlockComponent>
+      <Text>
+        <H5 bold>
+          <FormattedMessage id={`tables.State.title`} />
```

Comment:
For all places, if this is a fixed string, please use `id="tables.state.title"` and no template string.